File: InvoiceSync-be/src/main/java/com/invoicesync/shared/AccountingLineItem.java
package com.invoicesync.shared;

import lombok.Builder;

@Builder
public record AccountingLineItem(

    Long id,

    String name,

    Integer quantity,

    String unit,

    Integer vatRate,

    MonetaryAmount unitPrice,

    MonetaryAmount totalPrice,

    AccountingAssignment accounting

) {

}

File: InvoiceSync-fe/src/app/core/models/accounting-line-item.ts
import {MonetaryAmountDto} from "./monetary-amount-dto";
import {AccountingAssignment} from "./accounting-assignment";

export interface AccountingLineItem {

  id: number;
  name: string;
  quantity: number;
  unit: string;
  vatRate: number;
  unitPrice: MonetaryAmountDto;
  totalPrice: MonetaryAmountDto;
  accounting: AccountingAssignment;
}
